### PR TITLE
Adds support for class decorator as a minimal approach to #118

### PR DESCRIPTION
**Pull request by mikaelbr**
cc @torgeir

### CODE REVIEW

File: component.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ module.exports = factory();
  *   shouldComponentUpdate: function(nextProps, nextState), // check update
  *   cursorField: '__singleCursor', // cursor property name to "unwrap" before passing in to render
  *   isNode: function(propValue), // determines if propValue is a valid React node
+ *   classDecorator: function(Component), // Allows for decorating created class
```
Comment (torgeir) [inline component.js]: Will we be decorating other stuff than the class, or is `decorator` enough? Maybe.. don't know what that would be at the moment, though

File: component.js [hunk 2]
```diff
@@ -83,15 +103,16 @@ module.exports = factory();
...
-  var _hiddenCursorField = options.cursorField || '__singleCursor';
-  var _isNode = options.isNode || isNode;
+  initialOptions = initialOptions || {};
+  var _shouldComponentUpdate = initialOptions.shouldComponentUpdate ||
+        shouldComponentUpdate.withDefaults(initialOptions);
+  var _isCursor = initialOptions.isCursor || shouldComponentUpdate.isCursor;
+  var _isImmutable = initialOptions.isImmutable || shouldComponentUpdate.isImmutable;
+  var _hiddenCursorField = initialOptions.cursorField || '__singleCursor';
+  var _isNode = initialOptions.isNode || isNode;
+  var _decorator = initialOptions.classDecorator || identity;
```
Comment (torgeir) [inline component.js]: If the other one is `classDecorator` this one should be as well.

File: component.js [hunk 3]
```diff
@@ -141,7 +162,7 @@ function factory (options) {
       removeOldStaticMethods(options.mixins);
     }
 
-    var Component = React.createClass(componentObject);
+    var Component = _decorator(React.createClass(componentObject));
```
Comment (torgeir) [inline component.js]: I like it.

File: tests/component-test.js [hunk 4]
```diff
@@ -42,6 +42,44 @@ describe('component', function () {
...
+    it('should attach decorator', function () {
+      var hasBeenCalled = false;
+      var decorator = function (SomeClass) {
+        hasBeenCalled = true;
+        return SomeClass;
+      };
+      var decoratedComponent = component.withDefaults({
+        classDecorator: decorator
+      });
+      var Component = decoratedComponent(function MyComponentName () {
```
Comment (torgeir) [inline tests/component-test.js]: Newline above this one.

File: tests/component-test.js [hunk 5]
```diff
@@ -42,6 +42,44 @@ describe('component', function () {
...
+    it('should allow to extend class as decorator', function () {
+      var decorator = function (ComposedComponent) {
+        ComposedComponent.displayName.should.equal('MyComponentName');
+        ComposedComponent.displayName = 'Foobar';
+        return ComposedComponent;
+      };
+      var decoratedComponent = component.withDefaults({
+        classDecorator: decorator
+      });
+      var Component = decoratedComponent(function MyComponentName () {
```
Comment (torgeir) [inline tests/component-test.js]: Newline above this one.

**Comment by torgeir** [comment]
Nitpicks, looks good to me. Ought to fix and make #118 unnecessary

**Comment by torgeir** [comment]
@polkovnikov-ph

File: component.js (hunk 1 above)
Comment (mikaelbr) [inline component.js]: I think it's more clear to call it `classDecorator` to avoid any confusion.

**Comment by mikaelbr** [comment]
Thanks for your contribution and work! Appreciate it very much. We decided to go in a simpler direction to create factories that can create automatic decorated components in #121, but we wouldn't have gotten there without you!

**Pull request merged**